src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -281,6 +281,11 @@ internal void GetDeclarationDiagnostics(BindingDiagnosticBag addTo)
             GetAttributes();
             GetReturnTypeAttributes();
 
+            if (this.HasAsyncMethodBuilder(out _))
```

Comment:
>HasAsyncMethodBuilder [](http://example.com/codeflow?start=21&length=21)

It may be redundant, but consider checking `MessageID.IDS_AsyncMethodBuilderOverride.CheckFeatureAvailability()` here.

Reply:
After I merge with explicit return types that won't be needed, since you've added a call to `AsyncMethodChecks` here.